fsspec/caching.py

```diff
@@ -574,7 +574,7 @@ def __init__(
         blocksize: int,
         fetcher: Fetcher,
         size: int,
-        data: dict[tuple[int, int], bytes] = {},
+        data: dict[tuple[int, int], bytes] = None,
```

Comment:
The type hint is now wrong. It could be made into a class attribute hint instead...

Reply:
I've added `Optional[]` for consistency with the other arguments, which all have a type hint.